File: backend/semgrep_rules/tests/javascript/kisa_sw_002_good.js
function getStatus() {

  return "OK";
}


function getVersion() {

  return "1.0";
}


function execute(
  req
) {

  const action =
    req.query.action;

  const allowedActions = {
    status: getStatus,
    version: getVersion,
  };

  const handler =
    allowedActions[action];

  if (!handler) {

    throw new Error(
      "허용되지 않은 요청입니다."
    );
  }

  return handler();
}

File: backend/semgrep_rules/tests/python/kisa_sw_002_good.py
from flask import request


def get_status():

    return "OK"


def get_version():

    return "1.0"


def execute():

    action = request.args.get(
        "action"
    )

    allowed_actions = {
        "status": get_status,
        "version": get_version,
    }

    handler = allowed_actions.get(
        action
    )

    if handler is None:

        raise ValueError(
            "허용되지 않은 요청입니다."
        )

    return handler()
